client_functions.go

```diff
@@ -91,8 +102,8 @@ func (c *Client) ThermostatSummary() (*ThermostatSummary, error) {
 	}
 
 	ts := &ThermostatSummary{}
-	if err := json.NewDecoder(res.Body).Decode(ts); err != nil {
-		return nil, err
+	if err := jsonDecode(res.Body, ts); err != nil {
+		return nil, fmt.Errorf("failed to decode response from API: %v", err)
```

Comment:
nit: Formatting error that was already formatted in original function with no added value.

Reply:
Removed.